core/frontend/src/components/wifi/ConnectionDialog.vue

```diff
@@ -164,24 +215,28 @@ export default Vue.extend({
     },
     async connectToWifiNetwork(): Promise<void> {
       const credentials: NetworkCredentials = { ssid: this.real_ssid, password: this.password }
+      this.connection_status = ConnectionStatus.Connecting
+      this.connection_result_message = ''
       await back_axios({
         method: 'post',
         url: `${wifi.API_URL}/connect`,
-        timeout: 2000,
+        timeout: 20000,
         data: credentials,
         params: { hidden: this.is_hidden },
       })
         .then(() => {
+          this.connection_status = ConnectionStatus.Succeeded
+          this.connection_result_message = `Successfully connected to '${this.real_ssid}'`
           wifi.setNetworkStatus(null)
+          this.password = ''
+          this.force_password = false
         })
         .catch((error) => {
-          const message = `Could not connect to wifi network: ${error.message}.`
+          this.connection_status = ConnectionStatus.Failed
+          const message = error.response.data.detail ? error.response.data.detail : `Connection failed: ${error}`
+          this.connection_result_message = message
```

Comment:
We could add a message saying to check if the wifi password is correct.

Reply:
Definitely. Adding.